auth/src/main/java/com/firebase/ui/auth/compose/FirebaseAuthUI.kt

```diff
@@ -56,6 +62,154 @@ class FirebaseAuthUI private constructor(
+     *
+     * **Example:**
+     * ```kotlin
+     * val authUI = FirebaseAuthUI.getInstance()
+     * val user = authUI.getCurrentUser()
+     * user?.let {
+     *     println("User email: ${it.email}")
+     *     println("User ID: ${it.uid}")
+     * }
+     * ```
+     *
+     * @return The currently signed-in [FirebaseUser], or `null` if no user is signed in
+     */
+    fun getCurrentUser(): FirebaseUser? = auth.currentUser
+
+    /**
+     * Returns a [Flow] that emits [AuthState] changes.
+     *
+     * This flow observes changes to the authentication state and emits appropriate
+     * [AuthState] objects. The flow will emit:
+     * - [AuthState.Idle] when there's no active authentication operation
+     * - [AuthState.Loading] during authentication operations
+     * - [AuthState.Success] when a user successfully signs in
+     * - [AuthState.Error] when an authentication error occurs
+     * - [AuthState.Cancelled] when authentication is cancelled
+     * - [AuthState.RequiresMfa] when multi-factor authentication is needed
+     * - [AuthState.RequiresEmailVerification] when email verification is needed
+     *
+     * The flow automatically emits [AuthState.Success] or [AuthState.Idle] based on
+     * the current authentication state when collection starts.
+     *
+     * **Example:**
+     * ```kotlin
+     * val authUI = FirebaseAuthUI.getInstance()
+     *
+     * lifecycleScope.launch {
+     *     authUI.authStateFlow().collect { state ->
+     *         when (state) {
+     *             is AuthState.Success -> {
+     *                 // User is signed in
+     *                 updateUI(state.user)
+     *             }
+     *             is AuthState.Error -> {
+     *                 // Handle error
+     *                 showError(state.exception.message)
+     *             }
+     *             is AuthState.Loading -> {
+     *                 // Show loading indicator
+     *                 showProgressBar()
+     *             }
+     *             // ... handle other states
+     *         }
+     *     }
+     * }
+     * ```
+     *
+     * @return A [Flow] of [AuthState] that emits authentication state changes
+     */
+    fun authStateFlow(): Flow<AuthState> = callbackFlow {
+        // Set initial state based on current auth state
+        val initialState = auth.currentUser?.let { user ->
+            AuthState.Success(result = null, user = user, isNewUser = false)
+        } ?: AuthState.Idle
+
+        trySend(initialState)
+
+        // Create auth state listener
+        val authStateListener = AuthStateListener { firebaseAuth ->
+            val currentUser = firebaseAuth.currentUser
+            val state = if (currentUser != null) {
+                // Check if email verification is required
+                if (!currentUser.isEmailVerified &&
+                    currentUser.email != null &&
+                    currentUser.providerData.any { it.providerId == "password" }) {
```

Comment:
should we use `it.providerId == EmailAuthProvider.EMAIL_PASSWORD_SIGN_IN_METHOD` here instead, so it's clearer what is being compared?

Reply:
I think with the comments, it's pretty clear so no need to add extra checks